tests/models/test_hooks.py

```diff
@@ -235,6 +235,22 @@ def get_members(cls):
     return {h for h, _ in getmembers(cls, predicate=isfunction) if not h.startswith('_')}
 
 
+class HookedCallback(Callback):
```

Comment:
now that you are developing this api for hook call tracing, it could probably go into its own test utils file and then have tests (for its functionality).

Reply:
I don't like moving it to a different file since it's very little code and it's not used elsewhere. If some other file ends up needing it, then we can move it.

> and then have tests (for its functionality).

This is not necessary because once you manually assert the correctness of the first test, that test becomes also the test for functionality (since it would fail otherwise)